source/SVInt.cpp

```diff
@@ -476,6 +476,71 @@ SVInt SVInt::lshr(uint32_t amount) const {
     return result;
 }
 
+SVInt SVInt::ashr(const SVInt& rhs) const {
+    if (!signFlag)
+        return lshr(rhs);
+    if (rhs.hasUnknown())
+        return createFillX(bitWidth, signFlag);
+    if (rhs >= bitWidth)
+        return SVInt(bitWidth, *this >= 0 ? 0 : -1, signFlag);
+
+    uint16_t contractedWidth = bitWidth - rhs.getAssertUInt64();
+    SVInt tmp = lshr(rhs);
+    tmp.bitWidth = contractedWidth;
+    return signExtend(tmp, bitWidth);
+}
+
+SVInt SVInt::ashr(uint32_t amount) const {
+    if (!signFlag)
+        return lshr(amount);
+    if (amount == 0)
+        return *this;
+    if (amount >= bitWidth)
+        return SVInt(bitWidth, *this >= 0 ? 0 : -1, signFlag);
+
+    uint16_t contractedWidth = bitWidth - amount;
+    SVInt tmp = lshr(amount);
+    tmp.bitWidth = contractedWidth;
+    return signExtend(tmp, bitWidth);
+}
+
+SVInt SVInt::ambiguousConditionalCombination(const SVInt& rhs) const {
+    if (exactlyEqual(*this, rhs)) {
+        return rhs;
+    }
+    bool bothSigned = signFlag && rhs.signFlag;
+
+    if (bitWidth != rhs.bitWidth) {
+        if (bitWidth < rhs.bitWidth)
+            return extend(*this, rhs.bitWidth, bothSigned).ambiguousConditionalCombination(rhs);
+        else
+            return this->ambiguousConditionalCombination(extend(rhs, bitWidth, bothSigned));
+    }
+
+    if (isSingleWord()) {
+        // If the inputs are single word, as the inputs are unequal, we need to
+        // combine into a 2-word output with a word for unknowns
+        SVInt tmp = createFillX(bitWidth, bothSigned);
+        tmp.pVal[1] = val ^ rhs.val;
+        tmp.pVal[0] = ~tmp.pVal[1] & val & rhs.val;
+        return tmp;
+    }
+    SVInt tmp(*this);
```

Comment:
Creating this temporary is wasteful since you immediately delete the memory and allocate your own. There's a separate constructor that lets you just pass in allocated memory to use.